### PR TITLE
Added Validation for runtime types

### PR DESCRIPTION
**Pull request by PascalSenn**

### CODE REVIEW

File: src/HotChocolate/Core/src/Types/Types/Interceptors/TypeValidationTypeInterceptor.cs [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+
+        private void ValidateInputObjectType(
+            ITypeDiscoveryContext context,
+            InputObjectTypeDefinition? definition)
+        {
+            if (definition is { RuntimeType: { } runtimeType } &&
+                IsTypeSystemType(definition.RuntimeType))
+            {
+                ReportRuntimeTypeError(context, runtimeType);
+                definition.RuntimeType = typeof(Stub);
```
Comment (michaelstaib) [inline src/HotChocolate/Core/src/Types/Types/Interceptors/TypeValidationTypeInterceptor.cs:55]: why not just object?

**Review by michaelstaib — CHANGES_REQUESTED** [review]
I think the stub type can go, just use object there... you could even leave the runtime type since you are clearing the dependencies. I have to revisit the dependency clear again since I have reworked this part.

Also, we need an option that disables this interceptor. So, if the runtime check is disabled we should not include this interceptor.

**Comment by PascalSenn** [comment]
resolved

**Review by michaelstaib — APPROVED** [review]

**Pull request merged**